mixer/template/authz/template.proto

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package authz;
+
+import "mixer/v1/config/descriptor/value_type.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_CHECK;
+
+// List of subjects
+message Subject {
+  // The user.
+  string user = 1;
+  // Groups the subject belongs to.
+  repeated string groups = 2;
+  // Additional attributes about the subject.
+  map<string, istio.mixer.v1.config.descriptor.ValueType> extra = 3;
+}
+
+// List of actions
+message Action {
+  // Namespace the target action is taking place in.
+  string namespace = 1;
+  // The Service the action is being taken on.
+  string service = 2;
```

Comment:
I don't understand the context in which this information is needed. By the time the adapter gets a request, the notion of 'service' has really been erased from the data. Consider that when we report a metric, the metric object doesn't indicate the service the metric is associated with.

Reply:
is this the `destination.service` ?

Reply:
This will only be used in a Check request. "service" refers to destination service name.